lib/updater.rb

```diff
@@ -34,7 +32,10 @@ def edit_contacts(contact_index)
   def update_contacts
     loop do
       contact_index = user_interface.choose_contact(database.all)
-      edit_contacts(contact_index)
+      contact = database.contact_at(contact_index)
+
+      user_interface.display_contact(contact)
+      edit_contacts(contact_index, contact)
```

Comment:
Something to think about for the future (nothing to update in the code):
a more sensible way to do this is to pass the contact only, but the contact would have the index already as one of its fields. Then you don't have to pass both the contact and the index along, which is weird.

This happens because we are using an array as a database, but when we use a real database the contact will come with an id already.

(you could make the array behave like a database by appending the index to a contact as a field before returning it, but we won't do this yet)

Reply:
'a more sensible way to do this.....'  I've only done what you have asked me to do

Reply:
" I've only done what you have asked me to do"

I may be wrong but that sounds defensive. I am not criticising what you have done, I am explaining a concept.

This comment is to explain how a real database works, and how things are handled there, since things are handled differently to how arrays work. 

As I said at the beginning, "_Something to think about for the future (**nothing to update in the code**)_". In the future means, when we introduce a real database in the app.